test/extension/config/default/extension.yaml

```diff
@@ -19,6 +19,7 @@ spec:
         image: controller:latest
         name: extension
       terminationGracePeriodSeconds: 10
+      serviceAccountName: extension
```

Comment:
I would suggest to use `test-extension` to have all the objects named in a similar way (also applies to other files)